pennylane/qnode.py

```diff
@@ -592,6 +592,16 @@ def construct(self, args, kwargs):
 
     def __call__(self, *args, **kwargs):
 
+        for pos_arg in args:
+            if not isinstance(pos_arg, qml.numpy.tensor):
+                warnings.warn(
+                    "At least one of the positional arguments passed to the QNode"
+                    "is not a PennyLane NumPy tensor, though it is regarded as a trainable parameter. "
+                    "Please revisit the positional arguments when calling the QNode and pass non-tensor "
+                    "arguments as keyword arguments when calling the QNode.",

```

Comment:
Suggest rewording this to clarify the change, and avoid mentions of tensors. Something like the following:

```python
f"""VisibleDeprecationWarning: Passing non-differentiable positional arguments to
QNodes without setting requires_grad=False has been deprecated and will be
removed from a future release.

Please specify directly that this argument is non-trainable via np.array({pos_arg}, requires_grad=False),
or for non-numeric arguments consider passing it as a keyword argument."""
```